ala-name-matching-builder/src/main/java/au/org/ala/names/index/ALANameAnalyser.java

```diff
@@ -299,7 +299,18 @@ public NameKey analyse(@Nullable NomenclaturalClassifier code, String scientific
         scientificName = scientificName.trim().toUpperCase();
 
 
-        return new NameKey(this, code, scientificName, scientificNameAuthorship, rankType, nameType, flags);
+        NameKey key = new NameKey(this, code, scientificName, scientificNameAuthorship, rankType, nameType, flags);
+        if (name == null)
```

Comment:
Reminder: un-parentheses `if/else` syntax is discouraged according to our coding std ([oracle standard](https://www.oracle.com/java/technologies/javase/codeconventions-statements.html#449))

Reply:
Fixed with change to logic